### PR TITLE
bugfix/LIVE-5364: Fix bug in device language change prompt

### PR DESCRIPTION
**Pull request by juan-cortes**
### 📝 Description

We were not storing the changed language on the last seen device information leaving to a missed opportunity to prompt the user to also change the device language when they changed the app language and we can offer it. It's easier to understand with the test instructions really.

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` 
- **Linked resource(s)**:  https://ledgerhq.atlassian.net/browse/LIVE-5364 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://user-images.githubusercontent.com/4631227/219421109-18a0d5a4-49ed-4aa3-a6b5-ec5660668b3f.mp4

### 🚀 Expectations to reach
- LLM should remember the last seen device language now.
- Changing LLM to a language supported by FW should prompt the user.
- If the last seen device language is already LLM's language, no prompt.

### CODE REVIEW

File: apps/ledger-live-mobile/src/reducers/settings.ts
Comment (alexandremgo) [inline apps/ledger-live-mobile/src/reducers/settings.ts:478]: I think it's normal to have this check on `lastSeenDevice`, because the action's `payload` is only a `Partial<DeviceInfo>` -> so if `lastSeenDevice` does not exist, we would have a `deviceInfo: Partial<DeviceInfo>` and not `deviceInfo: DeviceInfo` ;)

**Review by alexandremgo — APPROVED** [review]
👌

File: apps/ledger-live-mobile/src/reducers/settings.ts
Comment (juan-cortes) [reply apps/ledger-live-mobile/src/reducers/settings.ts:478]: So it's not that I didn't want to make it happy, I just couldn't because of the nature of our relationship. Got it.

Comment (juan-cortes) [reply apps/ledger-live-mobile/src/reducers/settings.ts:478]:
```suggestion
    if (!state.lastSeenDevice) return state;
```

**Review by ofreyssinet-ledger — APPROVED** [review]
lgtm 👍

File: apps/ledger-live-mobile/src/components/ModalLock.tsx
Comment (alexandremgo) [inline apps/ledger-live-mobile/src/components/ModalLock.tsx:24]: Should it be memoized to avoid re-rendering it when it's not necessary ? This will reduce the number of dispatch occurring

Comment (juan-cortes) [reply apps/ledger-live-mobile/src/components/ModalLock.tsx:24]: I tested it (as per the video) and it didn't seem to cause any problem. Worst case scenario is we unlock and lock again.
I think we can live with it

Comment (alexandremgo) [reply apps/ledger-live-mobile/src/components/ModalLock.tsx]: Ok !

**Comment by juan-cortes** [comment]
Waiting for CI to green light this since it's QA'ed and reviewed

**Pull request merged**
